array bounds overflow in transport_ofi.c

```
transport_ofi.c:842:5: warning: array index 256 is past the end of the array
      (which contains 256 elements) [-Warray-bounds]
    myephostname[EPHOSTNAMELEN] = '\0';
    ^            ~~~~~~~~~~~~~
transport_ofi.c:83:1: note: array 'myephostname' declared here
static char         myephostname[EPHOSTNAMELEN];
^
1 warning generated.
```

(BTW detected when compiling with clang)

--- a/src/transport_ofi.c
+++ b/src/transport_ofi.c
@@ -830,7 +830,7 @@ static inline int publish_av_info(struct fabric_info *info)
     if(gethostname(myephostname, (EPHOSTNAMELEN - 1)) != 0)
         OFI_ERRMSG("gethostname error: %s \n", strerror(errno));
 
-    myephostname[EPHOSTNAMELEN] = '\0';
+    myephostname[EPHOSTNAMELEN-1] = '\0';
 
     ret = shmem_runtime_put("fi_ephostname", myephostname, EPHOSTNAMELEN);
     if (ret != 0) {

